### PR TITLE
build(deps): bump lint_version from 27.2.2 to 30.0.0

### PR DESCRIPTION
**Pull request by david-allison-1**
Fix errors which were introduced:

* api is now required
* Unicode Handling changed
* Compilation errors from missing imports were introduced

Bumps `lint_version` from 27.2.2 to 30.0.0.

Updates `lint-api` from 27.2.2 to 30.0.0

Updates `lint` from 27.2.2 to 30.0.0

Updates `lint-tests` from 27.2.2 to 30.0.0

---
updated-dependencies:
- dependency-name: com.android.tools.lint:lint-api
  dependency-type: direct:production
  update-type: version-update:semver-major
- dependency-name: com.android.tools.lint:lint
  dependency-type: direct:production
  update-type: version-update:semver-major
- dependency-name: com.android.tools.lint:lint-tests
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

### CODE REVIEW

**Comment by david-allison-1** [comment]
Suprsedes #9327

File: lint-rules/src/test/java/com/ichi2/anki/lint/rules/CopyrightHeaderExistsTest.java [hunk 1]
```diff
@@ -73,6 +73,7 @@ public void fileWithCopyrightHeaderPasses() {
     public void fileWithNoCopyrightHeaderFails() {
         lint()
                 .allowMissingSdk()
+                .allowCompilationErrors() // import failures
```
Comment (mikehardy) [inline lint-rules/src/test/java/com/ichi2/anki/lint/rules/CopyrightHeaderExistsTest.java:76]: I see

> Compilation errors from missing imports were introduced

Can elaborate just a little bit? Just a link to where-ever this is documented would work for me

**Review by mikehardy — APPROVED** [review]
thanks for taking this on! saves me time for sure - I don't consider asking for a link any sort of a disapproval so merge at well after chucking a link in somewhere

File: lint-rules/src/test/java/com/ichi2/anki/lint/rules/CopyrightHeaderExistsTest.java (hunk 1 above)
Comment (david-allison-1) [reply lint-rules/src/test/java/com/ichi2/anki/lint/rules/CopyrightHeaderExistsTest.java:76]:
```
java.lang.IllegalStateException: 
app\src\com\ichi2\upgrade\Upgrade.java:4: Error: 
Couldn't resolve this import [LintError]
import com.ichi2.libanki.Collection;
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

This usually means that the unit test needs to declare a stub file or
placeholder with the expected signature such that type resolving works.

If this import is immaterial to the test, either delete it, or mark
this unit test as allowing resolution errors by setting
`allowCompilationErrors()`.

(This check only enforces import references, not all references, so if
it doesn't matter to the detector, you can just remove the import but
leave references to the class in the code.)

For more information, see the "Library Dependencies and Stubs" section in
https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-master-dev:lint/docs/api-guide/unit-testing.md.html

	at com.android.tools.lint.checks.infrastructure.ResolveCheckerKt$checkFile$1.reportResolveProblem(ResolveChecker.kt:183)
	at com.android.tools.lint.checks.infrastructure.ResolveCheckerKt$checkFile$1.visitImportStatement(ResolveChecker.kt:105)
	at org.jetbrains.uast.UImportStatement$DefaultImpls.accept(UImportStatement.kt:41)
	at org.jetbrains.uast.java.JavaUImportStatement.accept(JavaUImportStatement.kt:25)
	at org.jetbrains.uast.internal.ImplementationUtilsKt.acceptList(implementationUtils.kt:23)
	at org.jetbrains.uast.UFile$DefaultImpls.accept(UFile.kt:82)
	at org.jetbrains.uast.java.JavaUFile.accept(JavaUFile.kt:11)
	at com.android.tools.lint.checks.infrastructure.ResolveCheckerKt.checkFile(ResolveChecker.kt:62)
	at com.android.tools.lint.checks.infrastructure.TestLintClient$2.parse(TestLintClient.java:880)
	at com.android.tools.lint.client.api.UElementVisitor.visitFile(UElementVisitor.kt:240)
	at com.android.tools.lint.client.api.LintDriver$visitUastDetectors$1.run(LintDriver.kt:2019)
	at com.android.tools.lint.client.api.LintClient.runReadAction(LintClient.kt:1780)
	at com.android.tools.lint.checks.infrastructure.TestLintClient.runReadAction(TestLintClient.java:893)
	at com.android.tools.lint.client.api.LintDriver$LintClientWrapper.runReadAction(LintDriver.kt:2634)
	at com.android.tools.lint.client.api.LintDriver.visitUastDetectors(LintDriver.kt:2019)
	at com.android.tools.lint.client.api.LintDriver.visitUast(LintDriver.kt:1982)
	at com.android.tools.lint.client.api.LintDriver.runFileDetectors(LintDriver.kt:1390)
	at com.android.tools.lint.client.api.LintDriver.checkProject(LintDriver.kt:1151)
	at com.android.tools.lint.client.api.LintDriver.checkProjectRoot(LintDriver.kt:628)
	at com.android.tools.lint.client.api.LintDriver.access$checkProjectRoot(LintDriver.kt:153)
	at com.android.tools.lint.client.api.LintDriver$analyze$1.invoke(LintDriver.kt:420)
	at com.android.tools.lint.client.api.LintDriver$analyze$1.invoke(LintDriver.kt:418)
	at com.android.tools.lint.client.api.LintDriver.doAnalyze(LintDriver.kt:502)
	at com.android.tools.lint.client.api.LintDriver.doAnalyze$default(LintDriver.kt:473)
	at com.android.tools.lint.client.api.LintDriver.analyze(LintDriver.kt:417)
	at com.android.tools.lint.checks.infrastructure.TestLintClient.analyze(TestLintClient.java:671)
	at com.android.tools.lint.checks.infrastructure.TestLintClient.analyze(TestLintClient.java:602)
	at com.android.tools.lint.checks.infrastructure.TestLintClient.checkLint(TestLintClient.java:349)
	at com.android.tools.lint.checks.infrastructure.TestLintRunner.checkLint(TestLintRunner.kt:229)
	at com.android.tools.lint.checks.infrastructure.TestLintRunner.run(TestLintRunner.kt:172)
	at com.android.tools.lint.checks.infrastructure.TestLintTask.run(TestLintTask.java:933)
	at com.ichi2.anki.lint.rules.CopyrightHeaderExistsTest.fileWithNoCopyrightHeaderFails(CopyrightHeaderExistsTest.java:78)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:43)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:82)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:73)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:71)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:220)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:53)



Process finished with exit code -1
```

Comment (david-allison-1) [reply lint-rules/src/test/java/com/ichi2/anki/lint/rules/CopyrightHeaderExistsTest.java:76]: I've updated the commit with a reference to the commit

**Comment by mikehardy** [comment]
Nice, totally works for me - thanks. Lint already passing so I'm going to hit the button

**Pull request merged**